src/ui/server/helpers/summary-lists/policy-and-export/policy-and-date-fields/index.ts

```diff
@@ -0,0 +1,34 @@
+import { FIELDS } from '../../../../content-strings/fields/insurance';
+import { FIELD_IDS } from '../../../../constants';
+import fieldGroupItem from '../../generate-field-group-item';
+import formatDate from '../../../date/format-date';
+import { ApplicationPolicyAndExport, SummaryListItemData } from '../../../../../types';
+
+const {
+  INSURANCE: {
+    POLICY_AND_EXPORTS: {
+      TYPE_OF_POLICY: { POLICY_TYPE },
+      CONTRACT_POLICY: { REQUESTED_START_DATE },
+    },
+  },
+} = FIELD_IDS;
+
+/**
+ * generatePolicyAndDateFields
+ * Create all policy and date fields and values for the Insurance - Type of policy govukSummaryList
+ * @param {Object} All submitted policy and export data
+ * @returns {Object} All policy and date fields and values in an object structure for GOVUK summary list structure
+ */
+const generatePolicyAndDateFields = (answers: ApplicationPolicyAndExport) => {
+  const fields = [
+    fieldGroupItem({
+      field: { id: POLICY_TYPE, ...FIELDS[POLICY_TYPE] },
+      data: answers,
+    }),
+    fieldGroupItem({ field: { id: REQUESTED_START_DATE, ...FIELDS.CONTRACT_POLICY[REQUESTED_START_DATE] } }, formatDate(answers[REQUESTED_START_DATE])),
```

Comment:
this is quite long.  Could we assign to constants instead and then put the constant in the function call?

Reply:
I agree - linting made it like this, it's a lot clearer with new lines, but eslint doesn't like that 😞 . I am thinking of creating a new helper function `getFieldById`, to use throughout these summary list functions so we can do this:

```js
fieldGroupItem({ field: getFieldById(FIELDS, REQUESTED_START_DATE) }, formatDate(answers[REQUESTED_START_DATE])),
```

Instead of this:

```js
fieldGroupItem({ field: { id: REQUESTED_START_DATE, ...FIELDS.CONTRACT_POLICY[REQUESTED_START_DATE] } }, formatDate(answers[REQUESTED_START_DATE])),
```

What do you think?

Reply:
I think that will be better.  Quite hard to read currently

Reply:
Updated all summary lists to use this new approach with new helper function, `getFieldById` which we could use in other places as well. The only summary list that doesn't use this is the quote summary list because the "titles" are structured differently to other summary lists, tech debt